// Copyright 2020 The Cross-Media Measurement Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package wfa.common;

option java_package = "org.wfanet.common";
option java_multiple_files = true;

// Parameters used to generate the distributed components of a shifted two-sided
// geometric distributed random variable.
message DistributedGeometricDistributionParams {
  // The number of contributors to the global random variable. At least 2.
  int64 num = 1;
  // The p (success ratio) parameter of the polya distribution. 0<p<1.
  double p = 2;
  // Offset to shift the generated PolyaDiff, It is also the threshold where we
  // truncate the individual Polya random numbers.
  // The result will be in the range of [0, shift_offset*2].
  int64 shift_offset = 3;
}

// For detail of the parameters, refer to "Dwork, C. and Roth, A., 2014. The
// algorithmic foundations of differential privacy. Foundations and Trends in
// Theoretical Computer Science, 9(3-4), pp.211-407."
message DifferentialPrivacyParams {
  double epsilon = 1;
  double delta = 2;
}
